src/modules/draft/draftApi.ts

```diff
@@ -11,142 +11,167 @@ import {
-};
-
-export const updateDraft = draft =>
+export const updateDraft = async (draft: UpdatedDraftApiType): Promise<DraftApiType> =>
   fetchAuthorized(`${baseUrl}/${draft.id}`, {
     method: 'PATCH',
     body: JSON.stringify(draft),
   }).then(resolveJsonOrRejectWithError);
 
-export const createDraft = draft =>
+export const createDraft = async (draft: NewDraftApiType): Promise<DraftApiType> =>
   fetchAuthorized(`${baseUrl}/`, {
     method: 'POST',
     body: JSON.stringify(draft),
   }).then(resolveJsonOrRejectWithError);
 
-export const fetchDraftHistory = (id, language) => {
+export const searchDrafts = async (query: DraftSearchQuery): Promise<DraftSearchResult> =>
+  fetchAuthorized(`${baseUrl}/search/`, {
+    method: 'POST',
+    body: JSON.stringify(query),
+  }).then(resolveJsonOrRejectWithError);
+
+export const cloneDraft = async (
+  id: number,
+  language?: string,
+  addCopyPostfixToArticleTitle: boolean = true,
+): Promise<DraftApiType> => {
+  const query = queryString.stringify({
+    language,
+    'copied-title-postfix': addCopyPostfixToArticleTitle,
+    fallback: true,
+  });
+  const url = `${baseUrl}/clone/${id}?${query}`;
+  return fetchAuthorized(url, { method: 'POST' }).then(resolveJsonOrRejectWithError);
+};
+
+export const fetchDraftHistory = async (id: number, language: string): Promise<DraftApiType[]> => {
   const query = queryString.stringify({ language });
   const url = language
     ? `${baseUrl}/${id}/history?${query}&fallback=true`
     : `${baseUrl}/${id}/history`;
   return fetchAuthorized(url).then(resolveJsonOrRejectWithError);
 };
 
-export const deleteLanguageVersion = (id, language) =>
+export const deleteLanguageVersion = async (id: number, language: string): Promise<DraftApiType> =>
   fetchAuthorized(`${baseUrl}/${id}/language/${language}`, {
     method: 'DELETE',
   }).then(resolveJsonOrRejectWithError);
 
-export const fetchNewArticleId = id => {
+export const fetchNewArticleId = async (id: number): Promise<number> => {
   const url = `${baseUrl}/external_id/${id}`;
   return fetchAuthorized(url).then(resolveJsonOrRejectWithError);
 };
 
-export const validateDraft = (id, draft) =>
+export const validateDraft = async (id: number, draft: UpdatedDraftApiType): Promise<number> =>
   fetchAuthorized(`${baseUrl}/${id}/validate/`, {
     method: 'PUT',
     body: JSON.stringify(draft),
   }).then(resolveJsonOrRejectWithError);
 
-export const updateStatusDraft = (id, status) =>
+export const updateStatusDraft = async (
+  id: number,
+  status: DraftStatusTypes,
+): Promise<DraftApiType> =>
   fetchAuthorized(`${baseUrl}/${id}/status/${status}`, {
     method: 'PUT',
   }).then(resolveJsonOrRejectWithError);
 
-export const fetchAgreements = query =>
-  fetchAuthorized(`${baseAgreementsUrl}?query=${query}`).then(resolveJsonOrRejectWithError);
+export const fetchTags = async (language: string): Promise<TagResult> => {
+  const query = queryString.stringify({ size: 7000, language });
```

Comment:
skumleser litt, men skjønner ikke hva denne er og hvor 7000 kommer fra.

Reply:
Trudde ikkje lenger vi brukte denne funksjonen når vi har støtte for tag-search.

Reply:
Det stemmer, har ikke sett at vi bruker den lenger direkte i koden, men så at redux bruker den for å populere artikler og tags så er en liiiten sjanse på at den fortsatt brukes et eller annet sted. Ved et større rewrite og fjerning av redux kan vi få det faktisk utforska of fjerne helt. Foreløpig lar jeg den stå, just in case den det fortsatt brukes et eller annet sted.